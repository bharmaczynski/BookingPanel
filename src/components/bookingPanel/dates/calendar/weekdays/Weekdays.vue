<template functional>
    <div class="weekdays">
        <div v-for="weekday in props.weekdays" :key="weekday" class="weekdays__weekday">
            <span class="weekdays__weekday-label font-size-s">{{ weekday }}</span>
        </div>
    </div>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';

@Component
export default class Weekdays extends Vue {
    @Prop({ required: true })
    weekdays!: string[];
}
</script>
<style lang="scss" scoped>
.weekdays {
    position: relative;
    z-index: 10;
    display: flex;
    justify-content: center;
    margin-top: 10px;
    padding: 10px;

    &__weekday {
        width: 14%;
        text-align: center;
    }

    &__weekday-label {
        color: $gray;
    }
}
</style>
