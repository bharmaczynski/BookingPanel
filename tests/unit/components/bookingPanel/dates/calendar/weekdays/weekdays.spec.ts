import { shallowMount } from '@vue/test-utils';
import Weekdays from '@/components/bookingPanel/dates/calendar/weekdays/Weekdays.vue';

describe('Weekdays', () => {
    let propsData = {
        weekdays: ['aaaa', 'bb', 'cc', 'dd'],
    };

    function createWrapper(propsData = {}) {
        return shallowMount(Weekdays, {
            propsData,
        });
    }

    describe('Snapshots', () => {
        it('should match initial snapshot', () => {
            const wrapper = createWrapper(propsData);

            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
