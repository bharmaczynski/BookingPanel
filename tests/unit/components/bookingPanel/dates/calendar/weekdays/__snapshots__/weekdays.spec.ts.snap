// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekdays Snapshots should match initial snapshot 1`] = `
<div class="weekdays">
  <div class="weekdays__weekday"><span class="weekdays__weekday-label font-size-s">aaaa</span></div>
  <div class="weekdays__weekday"><span class="weekdays__weekday-label font-size-s">bb</span></div>
  <div class="weekdays__weekday"><span class="weekdays__weekday-label font-size-s">cc</span></div>
  <div class="weekdays__weekday"><span class="weekdays__weekday-label font-size-s">dd</span></div>
</div>
`;
